### PR TITLE
Add `src/components` folder (vue single file components refactor)

### PR DESCRIPTION
**Pull request by Raruto**
**List of changes:**

- add `src/components` folder
- refactor all vue components to use SFC
- move all SFC into `src/components` folder
- create `gui/vue/directives` folder
- move some custom vue directivies into `gui/vue/vue.directives.js`
- move some custom vue mixins into `gui/vue/vue.mixins.js`
- deprecate notice for `gui/inputs/select/vue/selectmixin.js`
- deprecate notice for `gui/metadata/vue/components/metadatamixin.js`
- fix issue on `gui/map/mapservice.js` map control (querybypolygon) when we click in area that no feature of selected layer is found. 

**TODO (before merge)**:

- test the fields within the `components/TableBody.vue` (eg. Right click on catalog element --> Open Attribute Table) see: https://github.com/g3w-suite/g3w-client/pull/83/commits/1969162315290d298ce0fa1cb2227cafcad0664a
- check updated `v-key` previously used on some `<template>` tags
- ~~bump version to v3.5 ?~~

### CODE REVIEW

File: src/components/Viewport.vue [hunk 1]
```diff
@@ -0,0 +1,215 @@
+<template>
```
Comment (Raruto) [inline src/components/Viewport.vue:5]: TODO: add a reference to original file location

File: src/services/viewport.js [hunk 2]
```diff
@@ -0,0 +1,631 @@
+import ApplicationState from 'core/applicationstate';
```
Comment (Raruto) [inline src/services/viewport.js:4]: TODO: add a reference to original file location

File: src/components/Floatbar.vue [hunk 3]
```diff
@@ -0,0 +1,67 @@
+<template>
```
Comment (Raruto) [inline src/components/Floatbar.vue:5]: TODO: add a reference to original file location

File: src/services/floatbar.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
+const {barstack:Stack} = require('gui/utils/utils');
```
Comment (Raruto) [inline src/services/floatbar.js:5]: TODO: add a reference to original file location

File: src/components/InputBase.vue [hunk 5]
```diff
@@ -0,0 +1,35 @@
+<template>
```
Comment (Raruto) [inline src/components/InputBase.vue:5]: TODO: add a reference to original file location

File: src/components/NavbaritemsLeft.vue [hunk 6]
```diff
@@ -0,0 +1,23 @@
+<template>
```
Comment (Raruto) [inline src/components/NavbaritemsLeft.vue:5]: TODO: add a reference to original file location

File: src/components/NavbaritemsRight.vue [hunk 7]
```diff
@@ -0,0 +1,22 @@
+<template>
```
Comment (Raruto) [inline src/components/NavbaritemsRight.vue:5]: TODO: add a reference to original file location

File: src/services/navbaritems.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
+const {base, inherit} = require('core/utils/utils');
```
Comment (Raruto) [inline src/services/navbaritems.js:4]: TODO: add a reference to original file location

File: src/components/Sidebar.vue [hunk 9]
```diff
@@ -0,0 +1,93 @@
+<template>
```
Comment (Raruto) [inline src/components/Sidebar.vue:5]: TODO: add a reference to original file location

File: src/components/SidebarItem.vue [hunk 10]
```diff
@@ -0,0 +1,83 @@
+<template>
```
Comment (Raruto) [inline src/components/SidebarItem.vue:5]: TODO: add a reference to original file location

File: src/services/sidebar.js [hunk 11]
```diff
@@ -0,0 +1,167 @@
+import SidebarItemComponent from 'components/SidebarItem.vue';
```
Comment (Raruto) [inline src/services/sidebar.js:4]: TODO: add a reference to original file location

**Review by Raruto — COMMENTED** [review]
@volterra79 please add some links or information related to this bug https://github.com/g3w-suite/g3w-client/pull/83/commits/f0ed0b9fca0a896931abf313caa9e2bef98e619e (otherwise it is difficult to understand what we are referring to)

**Comment by volterra79** [comment]
@Raruto solved https://github.com/g3w-suite/g3w-client/commit/f0ed0b9fca0a896931abf313caa9e2bef98e619e using slot="<name>" instead v-slot: <name>. I need to investigate in deep but i think this is a issue on vue-template-compire used by vueify (https://github.com/vuejs/vue/issues/9863) 
https://github.com/g3w-suite/g3w-client/pull/83/commits/e30ab7683effd355fc07efb9f611c63f12dec898

File: src/components/Field.vue [hunk 12]
```diff
@@ -1,3 +1,8 @@
+<!-- ORIGINAL SOURCE: -->
+<!-- gui/fields/field.vue@v3.4 -->
+
+<!-- FIXME: default content not displayed if no "slot" is provided (eg. see components/FieldText.vue) -->
+
```
Comment (Raruto) [inline src/components/Field.vue:5]: this file should be checked

**Review by volterra79 — APPROVED** [review]

**Pull request merged**